examples/train_model.py

```diff
@@ -101,8 +101,7 @@ def run_eval(agent, opt, datatype, max_exs=-1, write_log=False, valid_world=None
 
     if valid_world is None:
         valid_world = create_task(opt, agent)
-    else:
-        valid_world.reset()
+    valid_world.reset()
```

Comment:
is this the fix we were talking about before?

Reply:
this makes it so that the agent forgets any partial observations of training examples going into validation

Reply:
cool, yea that was the known bug that's in the TO DO, right?

Reply:
that's right! the custom metrics made it really obvious that there was carryover, because e.g. the lmloss in the language model (training loss) was being reported during the validation, and vice-versa. so these resets on both sides keep things clean, since the metrics are also reset.

Reply:
validation should finish episodes, though this does mean that training may have some examples truncated. that doesn't seem like a huge cost unless you're validating like every dozen examples though...